src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -473,8 +473,30 @@ static bool CallRegisterMethodByIndex (JniNativeMethodRegistrationArguments argu
 		public override void RegisterNativeMembers (JniType nativeClass, Type type, string? methods) =>
 			RegisterNativeMembers (nativeClass, type, methods.AsSpan ());
 
+#if ENABLE_MARSHAL_METHODS
+		// Temporary hack, see comments in RegisterNativeMembers below
+		static readonly Dictionary<string, List<string>> dynamicRegistrationMethods = new Dictionary<string, List<string>> (StringComparer.Ordinal) {
+			{"Android.Views.View+IOnLayoutChangeListenerImplementor",             new List<string> { "GetOnLayoutChange_Landroid_view_View_IIIIIIIIHandler" }},
+			{"Android.Views.View+IOnLayoutChangeListenerInvoker",                 new List<string> { "GetOnLayoutChange_Landroid_view_View_IIIIIIIIHandler" }},
+			{"Java.Interop.TypeManager+JavaTypeManager",                          new List<string> { "GetActivateHandler" }},
+		};
+#endif
+
 		public void RegisterNativeMembers (JniType nativeClass, Type type, ReadOnlySpan<char> methods)
 		{
+#if ENABLE_MARSHAL_METHODS
+			Logger.Log (LogLevel.Info, "monodroid-mm", $"RegisterNativeMembers ('{nativeClass?.Name}', '{type?.FullName}', '{methods.ToString ()}')");
+			Logger.Log (LogLevel.Info, "monodroid-mm", "RegisterNativeMembers called from:");
+			var st = new StackTrace (true);
+			foreach (string l in st.ToString ().Split ("\n")) {
```

Comment:
`Logger.Log()` *already* splits on newline: https://github.com/xamarin/xamarin-android/blob/903ba37ce70d2840983774e1d6fb55f8002561e2/src/Mono.Android/Android.Runtime/Logger.cs#L50

There is no need to split on `\n`.